### PR TITLE
Lock to veto permit

### PR DESCRIPTION
**Pull request by xavikh**

### CODE REVIEW

File: plugins/lockToVote/artifacts/ERC20withPermit.sol.ts
Comment (carlosgj94) [inline plugins/lockToVote/artifacts/ERC20withPermit.sol.ts:1]: This should be outside the lockToVote. It's a generic ERC20Permit. Also, change the name from ERC20withPermit to ERC20Permit

File: plugins/lockToVote/hooks/useProposalVeto.tsx [hunk 1]
```diff
@@ -73,12 +104,58 @@ export function useProposalVeto(proposalId: string) {
...
+
+    const domain = {
+      chainId: PUB_CHAIN.id,
+      name: String(erc20_name),
+      /* We assume 1 if permit version is not specified */
+      version: String(versionFromContract ?? '1'),
+      verifyingContract: PUB_TOKEN_ADDRESS,
+    };
+
+    const types = {
```
Comment (carlosgj94) [inline plugins/lockToVote/hooks/useProposalVeto.tsx:125]: Move this type to the types file in the /utils at the root of Aragonette

File: plugins/lockToVote/hooks/useProposalVeto.tsx [hunk 2]
```diff
@@ -73,12 +104,58 @@ export function useProposalVeto(proposalId: string) {
...
+
+    const message = {
+      owner: account_address,
+      spender: destination,
+      value,
+      nonce,
+      deadline,
+    };
+
+    vetoPermitSign({
```
Comment (carlosgj94) [inline plugins/lockToVote/hooks/useProposalVeto.tsx:143]: This part, plus the message could be in their own useHook at the root of Aragonette (not inside this plugin)

**Review by carlosgj94 — CHANGES_REQUESTED** [review]

File: plugins/lockToVote/hooks/useProposalVeto.tsx (hunk 1 above)
Comment (xavikh) [reply plugins/lockToVote/hooks/useProposalVeto.tsx]: I'm not sure if it makes sense in this case. It's not a TS type. It's just the Solidity type names to encode the permit message.

File: plugins/lockToVote/hooks/useProposalVeto.tsx [hunk 3]
```diff
@@ -70,14 +83,23 @@ export function useProposalVeto(proposalId: string) {
...
+    refetchPermitData();
   }, [vetoingStatus, vetoTxHash, isConfirming, isConfirmed]);
 
   const vetoProposal = () => {
-    vetoWrite({
-      abi: LockToVetoPluginAbi,
-      address: PUB_LOCK_TO_VOTE_PLUGIN_ADDRESS,
-      functionName: "veto",
-      args: [proposalId, 50000000000000000000],
+    let dest: `0x${string}` = PUB_LOCK_TO_VOTE_PLUGIN_ADDRESS;
```
Comment (carlosgj94) [inline plugins/lockToVote/hooks/useProposalVeto.tsx:90]:
```suggestion
    let dest: Address = PUB_LOCK_TO_VOTE_PLUGIN_ADDRESS;
```

Comment (carlosgj94) [reply plugins/lockToVote/hooks/useProposalVeto.tsx:90]: We would also need to import Address from wagmi/viem

**Review by carlosgj94 — CHANGES_REQUESTED** [review]

**Review by carlosgj94 — APPROVED** [review]

**Pull request merged**